src/on_demand/aca_on_demand_engine.cpp

```diff
@@ -63,18 +63,68 @@ ACA_On_Demand_Engine &ACA_On_Demand_Engine::get_instance()
   return instance;
 }
 
-OperationStatus ACA_On_Demand_Engine::unknown_recv(uint16_t vlan_id, string ip_src, string ip_dest,
-                                        int port_src, int port_dest, Protocol protocol) {
-  HostRequest HostRequest_builder;
-  HostRequest_ResourceStateRequest *new_state_requests = HostRequest_builder.add_state_requests();
-  HostRequestReply hostRequestReply;
+void ACA_On_Demand_Engine::process_async_grpc_replies()
+{
+  void *got_tag;
+  bool ok = false;
   HostRequestReply_HostRequestOperationStatus hostOperationStatus;
   OperationStatus replyStatus;
-  
-  uuid_t uuid;
-  uuid_generate_time(uuid);
-  char uuid_str[37];
-  uuid_unparse_lower(uuid, uuid_str);
+  std::unordered_map<std::__cxx11::string, data_for_on_demand_call *>::iterator found_data;
+  string uuid_for_call;
+  data_for_on_demand_call *data_for_uuid;
+  ACA_LOG_DEBUG("%s\n", "Beginning of process_async_grpc_replies");
+  while (cq_.Next(&got_tag, &ok)) {
+    if (ok) {
+      ACA_LOG_DEBUG("%s\n", "cq_->Next is good, ready to static cast the Async Client Call");
+
+      AsyncClientCall *call = static_cast<AsyncClientCall *>(got_tag);
+      ACA_LOG_DEBUG("%s\n", "Async Client Call casted successfully.");
+
+      if (call->status.ok()) {
+        ACA_LOG_DEBUG("%s\n", "Got an GRPC reply that is OK, need to process it.");
+        for (int i = 0; i < call->reply.operation_statuses_size(); i++) {
+          hostOperationStatus = call->reply.operation_statuses(i);
+          replyStatus = hostOperationStatus.operation_status();
+          uuid_for_call = hostOperationStatus.request_id();
+          found_data = request_uuid_on_demand_data_map.find(uuid_for_call);
+        }
+        ACA_LOG_DEBUG("Return from NCM - Reply Status: %s\n",
+                      to_string(replyStatus).c_str());
+        if (found_data != request_uuid_on_demand_data_map.end()) {
+          data_for_uuid = found_data->second;
+          ACA_LOG_DEBUG("Found data into the map, UUID: [%s], in_port: [%d], protocol: [%d]\n",
```

Comment:
@zzxgzgz is it better to replace `data_for_uuid` with `request_payload` or just `_payload` or `_data`?

Reply:
fixed